File: VB/HowToImplementDetailViewItem.Module.Win/WinModule.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.ComponentModel

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.NodeWrappers
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Model.NodeGenerators
Imports DevExpress.ExpressApp.Model.Core
Imports DevExpress.ExpressApp.Layout


Namespace HowToImplementDetailViewItem.Module.Win

	Public Class MyDetailViewLayoutUpdater
		Inherits ModelNodesGeneratorUpdater(Of ModelDetailViewLayoutNodesGenerator)
		Public Overrides Sub UpdateNode(ByVal node As DevExpress.ExpressApp.Model.Core.ModelNode)
			Dim layoutNode As IModelDetailViewLayout = CType(node, IModelDetailViewLayout)
			Dim myGroup As IModelLayoutGroup = layoutNode.AddNode(Of IModelLayoutGroup)("ClassIcon")
			myGroup.Index = 0
			myGroup.RelativeSize = 1
			myGroup.Direction = FlowDirection.Horizontal
			Dim myItem As IModelLayoutItem = myGroup.AddNode(Of IModelLayoutItem)("Icon")
		End Sub
	End Class

	Public Class MyDetailViewItemUpdater
		Inherits ModelNodesGeneratorUpdater(Of ModelDetailViewItemsNodesGenerator)
		Public Overrides Sub UpdateNode(ByVal node As DevExpress.ExpressApp.Model.Core.ModelNode)
			Dim itemsNode As IModelDetailViewItems = CType(node, IModelDetailViewItems)
			Dim myItem As IModelClassIcon = itemsNode.AddNode(Of IModelClassIcon)("Icon")
		End Sub
	End Class

	<ToolboxItemFilter("Xaf.Platform.Win")> _
	Public NotInheritable Partial Class HowToImplementDetailViewItemWindowsFormsModule
		Inherits ModuleBase
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Overrides Sub AddGeneratorUpdaters(ByVal updaters As ModelNodesGeneratorUpdaters)
			MyBase.AddGeneratorUpdaters(updaters)
			updaters.Add(New MyDetailViewLayoutUpdater())
			updaters.Add(New MyDetailViewItemUpdater())
		End Sub
	End Class
End Namespace

File: CS/HowToImplementDetailViewItem.Module.Win/WinModule.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.NodeWrappers;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Model.NodeGenerators;
using DevExpress.ExpressApp.Model.Core;
using DevExpress.ExpressApp.Layout;


namespace HowToImplementDetailViewItem.Module.Win {

    public class MyDetailViewLayoutUpdater : ModelNodesGeneratorUpdater<ModelDetailViewLayoutNodesGenerator> {
        public override void UpdateNode(DevExpress.ExpressApp.Model.Core.ModelNode node) {
            IModelViewLayout layoutNode = (IModelViewLayout)node;
            IModelLayoutGroup myGroup = layoutNode.AddNode<IModelLayoutGroup>("ClassIcon");            
            myGroup.Index = 0;
            myGroup.RelativeSize = 1;
            myGroup.Direction = FlowDirection.Horizontal;
            IModelLayoutItem myItem = myGroup.AddNode<IModelLayoutItem>("Icon");
        }
    }

    public class MyDetailViewItemUpdater : ModelNodesGeneratorUpdater<ModelDetailViewItemsNodesGenerator> {
        public override void UpdateNode(DevExpress.ExpressApp.Model.Core.ModelNode node) {
            IModelViewItems itemsNode = (IModelViewItems)node;
            IModelClassIcon myItem = itemsNode.AddNode<IModelClassIcon>("Icon");
        }
    }

	[ToolboxItemFilter("Xaf.Platform.Win")]
	public sealed partial class HowToImplementDetailViewItemWindowsFormsModule : ModuleBase {
		public HowToImplementDetailViewItemWindowsFormsModule() {
			InitializeComponent();
		}
        public override void AddGeneratorUpdaters(ModelNodesGeneratorUpdaters updaters) {
            base.AddGeneratorUpdaters(updaters);
            updaters.Add(new MyDetailViewLayoutUpdater());
            updaters.Add(new MyDetailViewItemUpdater());
        }
	}
}
